### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

- adds testing
- add watch tests command
- fix invalid field value bug
- fix find exception bug
 
**Type of Change**

- [ ] Bug
- [ ] Refactor
- [ ] New Feature
- [ ] Release Candidate

**Checklist before requesting review:**

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/submission/exceptions/exceptions.ts [hunk 1]
```diff
@@ -49,9 +49,9 @@ export const checkForProgramAndEntityExceptions = async ({
   schemaValidationErrors: dictionaryEntities.SchemaValidationError[];
 }) => {
   const filteredErrors: dictionaryEntities.SchemaValidationError[] = [];
-  let normalizedRecord = {};
+  let normalizedRecord = record;
```
Comment (justincorrigible) [inline src/submission/exceptions/exceptions.ts:52]: didn’t know typescript allowed for this simple definition 🤔

File: src/submission/exceptions/exceptions.ts [hunk 2]
```diff
@@ -141,9 +148,9 @@ export const validateFieldValueWithExceptions = ({
   let exceptionValue: string | undefined = '';
   // program level is applicable to ALL donors
   if (isProgramException(programException)) {
-    exceptionValue = programException.exceptions.find(exception => {
-      exception.requested_core_field === validationErrorFieldName;
```
Comment (justincorrigible) [inline src/submission/exceptions/exceptions.ts:145]: oh nice catch. seems like this ‘find’ was missing a return? 👍

File: test/unit/exception/applyException.spec.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { entities as dictionaryEntities } from '@overturebio-stack/lectern-client';
+import { SchemaValidationErrorTypes } from '@overturebio-stack/lectern-client/lib/schema-entities';
+import chai from 'chai';
+import sinon from 'sinon';
```
Comment (justincorrigible) [inline test/unit/exception/applyException.spec.ts:23]: fancy. hadn’t heard of this ☝️

File: src/exception/repo/program.ts [hunk 4]
```diff
@@ -35,10 +35,9 @@ const programExceptionSchema = new mongoose.Schema<ProgramException>({
   ],
 });
 
-const ProgramExceptionModel = mongoose.model<ProgramException>(
-  'ProgramException',
-  programExceptionSchema,
-);
+const ProgramExceptionModel =
+  mongoose.models.ProgramException ||
+  mongoose.model<ProgramException>('ProgramException', programExceptionSchema);
```
Comment (justincorrigible) [inline src/exception/repo/program.ts:40]: Not 100% sure what the difference is here. Is this a Mongoose nuance?

File: src/exception/util.ts [hunk 5]
```diff
@@ -22,11 +22,12 @@ import { EntityException, ProgramException, EntityValues, Entity } from './types
 import _ from 'lodash';
 
 export function isProgramException(result: ProgramException | null): result is ProgramException {
-  return (result as ProgramException).programId !== undefined;
+  return result !== null && result.programId !== undefined;
```
Comment (justincorrigible) [inline src/exception/util.ts:25]:
```suggestion
  return result?.programId !== undefined;
```

File: src/exception/repo/program.ts (hunk 4 above)
Comment (ciaranschutte) [reply src/exception/repo/program.ts:40]: this is a fun one. 
live watching for tests to re-run causes mongoose to try and reinitialize models, which it shouldn't do so it throws an error. this just makes sure these models haven't been init already, so no testing errors

Comment (justincorrigible) [reply src/exception/repo/program.ts:40]: Sounds like they have a bug or something. Maybe a comment here would be useful to explain intent?

**Review by justincorrigible — APPROVED** [review]
LGTM 👍

**Pull request merged**